BlueDBOnDisk/src/main/java/org/bluedb/disk/segment/Segment.java

```diff
@@ -166,6 +166,9 @@ public void modifyChunk(Path targetPath, StreamingWriter<T> processor) throws Bl
 
 	public void rollup(Range timeRange) throws BlueDbException {
 		rollup(timeRange, true);
+		if (getAllFileRangesInOrder(segmentPath).size() == 0) {

```

Comment:
If there are only duplicate files in the segment then the directory will not be deleted right?

Reply:
It shouldn't get deleted if there are any files with valid names in that folder.  Which I believe is the correct behavior.  We need the duplicates in there for queries even if there's nothing that is non-duplicate.